pytorch, M-Star and MARBLES; Q3 updates (#855)
NREL/HPC: docs/Documentation/Applications/LBMcfd.md

!!! note
    `HIPCC_COMPILE_FLAGS_APPEND` points `hipcc` to the GCC 14.2.0 toolchain loaded above. Without this, `hipcc` may pick up an incompatible system GCC and fail to compile device code.

!!! warning
    Do **not** set `AMREX_HOME` in your environment. MARBLES ships AMReX as a git submodule pinned to a specific commit that is tested and compatible with ROCm 7.2.0. Setting `AMREX_HOME` overrides this and will likely cause compile errors. If you have it set from another workflow, unset it before building:
    ```
    $ unset AMREX_HOME
    ```

Clone MARBLES and initialize the bundled AMReX submodule:
```
$ cd /projects/<projectname>/<username>/
$ mkdir marblesLBM && cd marblesLBM
$ git clone https://github.com/nileshsawant/marblesThermal
$ cd marblesThermal
$ git submodule update --init --recursive Submodules/AMReX
```

Build with HIP:
```
$ cd Build
$ make USE_HIP=TRUE COMP=clang USE_MPI=FALSE USE_RPATH=TRUE AMREX_AMD_ARCH=gfx90a
```

On success, the `Build` directory will contain the HIP executable, e.g.:
```
marbles3d.hip.TPROF.HIP.ex
```

Run a test case:
```
$ cp ../Tests/test_files/isothermal_cracks/* .
$ ./marbles3d.hip.TPROF.HIP.ex isothermal_cracks.inp
```

!!! note

>     For available modules on Gila, see the [Gila modules page](../Systems/Gila/modules.md). For partition details and node specifications, see the [Gila running page](../Systems/Gila/running.md).

Reviewer: ```suggestion
    See the [Gila Modules page](../Systems/Gila/modules.md) for information on the module system. For partition details and node specifications, see the [Running on Gila page](../Systems/Gila/running.md).
```

### M-Star on Gila (NVIDIA A100)

M-Star 4.1.15 is available on Gila's `gpu-intel-a100-80g` partition (8x A100-80GB GPUs per node, Intel Xeon Icelake CPUs). The module is installed under a separate module path that must be added before loading.

Request an interactive session on an A100 node:
```
$ salloc -A <projectname> -t 01:00:00 --nodes=1 --ntasks-per-node=42 --mem=800G --gres=gpu:8 --partition=gpu-intel-a100-80g
```

Load the M-Star module:
```

> $ module use /nopt/nrel/apps/modulefiles

Reviewer: ```suggestion
$ module load application
```

$ module load mstar/4.1.15
```

!!! note
    `mstar/4.1.15` automatically loads `cuda/13.1.1` and `hpcx/2.26` (OpenMPI). No separate module load commands are needed.

Run M-Star across all 8 GPUs on the node:
```
$ mpirun --map-by ppr:8:node:PE=1 -x PATH -x LD_LIBRARY_PATH -np 8 \
    mstar-cfd-mgpu -i input.xml -o out --gpu-ids "0,1,2,3,4,5,6,7"
```

An equivalent batch script:
```
#!/bin/bash
#SBATCH --account=<projectname>
#SBATCH --time=01:00:00
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=42
#SBATCH --mem=800G
#SBATCH --gres=gpu:8
#SBATCH --partition=gpu-intel-a100-80g
#SBATCH --job-name=mstar_gila
#SBATCH --output=slurm-%j.out

> module use /nopt/nrel/apps/modulefiles

Reviewer: ```suggestion
module load application
```
